lib/wallet/walletdb.js

```diff
@@ -1117,7 +1118,9 @@ class WalletDB extends EventEmitter {
     assert(data);
 
     const account = Account.fromRaw(this, data);
+    const childIndex = account.accountKey.childIndex;
 
+    account.accountPath = childIndex ^ HDCommon.HARDENED;
```

Comment:
I _think_ the account key's child index is already hardened, and therefore already > `0x80000000` 
so I think this xor is redundant

Reply:
This returns the expected value for hardened keys, which should always be the case with xpubs. (eg, returns accountPath '1' instead of '2147483649`). Perhaps there are edge cases of non-hardened xpubs, but not sure how to handle that one...

Reply:
Oh I see its an `xor` not `or` so the constant gets cancelled out 👍